superset-frontend/src/dashboard/containers/Chart.jsx

```diff
@@ -73,6 +72,11 @@ function mapStateToProps(
     sliceId: id,
     nativeFilters,
     dataMask,
+    labelColors:
+      dashboardInfo?.metadata?.label_colors &&
+      Object.keys(dashboardInfo.metadata.label_colors).length > 0
+        ? dashboardInfo.metadata.label_colors
+        : undefined,
```

Comment:
nit: as we're already using lodash here, maybe we could simplify some more and use [size](https://lodash.com/docs/4.17.15#size) for this:
```
    labelColors:
      size(dashboardInfo?.metadata?.label_colors)
        ? dashboardInfo.metadata.label_colors
        : undefined,
```